Tasks/NuGetCommandV2/nugetpublisher.ts

```diff
@@ -414,9 +417,12 @@ function shouldUseVstsNuGetPush(isInternalFeed: boolean, conflictsAllowed: boole
     return false;
 }
 
-function getAccessToken(isInternalFeed: boolean): string{
-    let accessToken: string;
+async function getAccessToken(isInternalFeed: boolean): Promise<string>{
     let allowServiceConnection = tl.getVariable('PUBLISH_VIA_SERVICE_CONNECTION');
+    let accessToken: string;
+
+    const collectionUrl = tl.getVariable("System.TeamFoundationCollectionUri");
+    const collectionPathName:string = url.parse(collectionUrl).pathname;
```

Comment:
This doesn't contain the collection name in the collectionname.visualstudio.com case (but see below for the solution using the discovered prefix list, which will have both forms).